lib/pubsub/iam.js

```diff
@@ -0,0 +1,190 @@
+ * @param {array=} policy.bindings - Bindings associate members with roles.
+ * @param {array=} policy.rules - Rules to be applied to the policy.
+ * @param {string=} policy.etag - Etags are used to perform a read-modify-write.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {object} callback.policy - The updated policy.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @alias iam.setPolicy
+ *
+ * @example
+ * var myPolicy = {
+ *   bindings: [{
+ *     role: 'roles/pubsub.subscriber',
+ *     members: ['serviceAccount:myotherproject@appspot.gserviceaccount.com']
+ *   }]
+ * };
+ *
+ * topic.iam.setPolicy(myPolicy, function(err, policy) {});
+ *
+ * subscription.iam.setPolicy(myPolicy, function(err, policy) {});
+ */
+IAM.prototype.setPolicy = function(policy, callback) {
+  if (!is.object(policy)) {
+    throw new Error('A policy is required');
+  }
+
+  var path = this.resource + ':setIamPolicy';
+  var body = {
+    policy: policy
+  };
+
+  this.makeReq_('POST', path, null, body, function(err, resp) {
+    callback(err, resp, resp);
+  });
+};
+
+/**
+ * Test a set of permissions for a resource
+ * Permissions with wildcards (such as '*' or 'storage.*') are not
+ * allowed
+ *
+ * @throws {Error} If permissions are not provided.
+ *
+ * @param {string|array} permissions - The permission(s) to test for.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {array} callback.permissions - A subset of permissions that the caller
+ * @param {object} callback.apiResponse - The full API response.
+ * is allowed
+ *
+ * @alias iam.testPermissions
+ *
+ * @example
+ * //-
+ * // Test a single permission
+ * //-
+ * var test = 'pubsub.topics.update';
+ *
+ * topic.iam.testPermissions(test, function(err, permissions, apiResponse) {
+ *   console.log(persmissions);
+ *   // {
+ *   //   "pubsub.topics.update": true
+ *   // }
+ * });
+ *
+ * //-
+ * // Or test several permissions at once
+ * //-
+ * var tests = [
+ *   'pubsub.subscriptions.consume',
+ *   'pubsub.subscriptions.update'
+ * ];
+ *
+ * subscription.iam.testPermissions(tests, function(err, permissions) {
+ *   console.log(permissions);
+ *   // {
+ *   //   "pubsub.subscriptions.consume": true,
+ *   //   "pubsub.subscriptions.update": false
+ *   // }
+ * });
+ */
+IAM.prototype.testPermissions = function(permissions, callback) {
+  if (!is.array(permissions) && !is.string(permissions)) {
+    throw new Error('Permissions are required');
+  }
+
+  var path = this.resource + ':testIamPermissions';
+  var body = {
+    permissions: arrify(permissions)
+  };
+
+  this.makeReq_('POST', path, null, body, function(err, resp) {
+    if (err) {
+      callback(err, null, resp);
+      return;
+    }
+
+    var availablePermissions = arrify(resp && resp.permissions);
```

Comment:
I believe this is always an array. And `resp` will always exist. Maybe easier to do `resp.permissions || []` if we're uncertain that it will be an array.